src/include/wiredtiger.in

```diff
@@ -770,6 +770,11 @@ struct __wt_session {
 	 */
 	void *app_private;
 #endif
+        /*!
+	 * A location for applications to attach a string that will be printed with verbose
+	 * messages.
+	 */
+        const char *verbose_private;
```

Comment:
I don't see any reason to make this public -- the application already has `WT_SESSION.app_private` and the the session is passed to verbose message handlers.

Reply:
Any user that wants to use verbose messages could now automatically have this string as part of the message. So it seemed reasonable to make it as public as `verbose` is.

Reply:
I believe this adds no new functionality over `WT_SESSION.app_private`, so I'd argue against it on those grounds. I'm ready to be wrong, if I'm just missing something this gives you that can't be done with `app_private`.

Reply:
It allows WT to incorporate that string into the middle of the verbose message in general. However, with `WT_VERB_DEBUG_8392` it means that using `app_private` won't interfere with any other application, which was @ddanderson 's concern. I'm happy to go back to that since it shouldn't interfere with any other verbose setting.

Reply:
So `app_private` won't be printed unless `WT_VERB_DEBUG_8392` is enabled?  As long as an application can put a non-string value into `app_private` and not have it dereferenced as a string, I'm happy.

Reply:
I'm kind of confused. Yeah, we could couple adding `WT_VERB_DEBUG_8392` to `app_private`, but if we're going to do that, isn't it simpler to just add `WT_SESSION.verb_8392`, and leave it as a private field?

Reply:
By 'private_field' do you mean under HAVE_DIAGNOSTIC? That ends up being kind of messy in `session_api.c:open_session` where we initialize the fields. It isn't hard, just kind of ugly.

Reply:
Ok, ignore my rerequest for a review because I left it in WT_SESSION. I'll move it to the WT_SESSION_IMPL and then you both can take a quick look.